lms/djangoapps/courseware/courses.py

```diff
@@ -161,6 +164,11 @@ def _check_nonstaff_access():
             if not enrollment_access_response:
                 return enrollment_access_response
 
+        if check_if_dsc_required:
+            data_sharing_consent_response = check_data_sharing_consent(course)
+            if not data_sharing_consent_response:
+                return data_sharing_consent_response
```

Comment:
Do we need to handle this response type scenario in `check_course_access_with_redirect()`?

Reply:
No, We want to add this access control in one API, that API is responsible for getting course page metadata.

check_course_access_with_redirect is mainly used in views, not API.